import { DomainException } from "../../../shared/exception/domain.exception";

class NomeInvalido extends DomainException{
  constructor(message: string = 'Nome Inv치lido'){
    super(message);
    this.message = message;
  }
}

class CPFinvalido extends DomainException{
  constructor(message: string = 'CPF inv치lido'){
    super(message);
    this.message = message;
  }
}

class IdadeInvalida extends DomainException{
  constructor(message: string = 'Idade inv치lida'){
    super(message);
    this.message = message;
  }
}

class TelefoneInvalido extends DomainException{
  constructor(message: string = 'Telefone inv치lido'){
    super(message);
    this.message = message;
  }
}

export const PacienteExceptions = {
  NomeInvalido,
  CPFinvalido,
  IdadeInvalida,
  TelefoneInvalido
}
